server/neptune/workflows/internal/deploy/revision/queue/deployer.go

```diff
@@ -86,12 +91,48 @@ func (p *Deployer) Deploy(ctx workflow.Context, requestedDeployment terraformWor
 		logger.Error(ctx, "unable to persist deployment, proceeding with in-memory value", key.ErrKey, persistErr)
 	}
 
+	// rebase open PRs
+	// worker uses the returned error types to perform some follow up tasks, so instead of propagating the rebase error back, we configure
+	// infinite retries and log the error to ensure we rebase open PRs before deploying another change in this root
+	// ScheduleToCloseTimeout is configured to 8 hours to ensure we're not blocking deploys for this root if the issue persists
+	if rebaseErr := p.rebaseOpenPRsForRoot(ctx, requestedDeployment.Repo); rebaseErr != nil {
+		scope.Counter("rebase_open_prs_err").Inc(1)
+		logger.Error(ctx, "unable to rebase open PRs", key.ErrKey, rebaseErr)
+	}
+
 	// Count this as deployment as latest if it's not a PlanRejectionError which means it is a TerraformClientError
 	// We do this as a safety measure to avoid deploying out of order revision after a failed deploy since it could still
 	// mutate the state file
 	return latestDeployment, err
 }
 
+func (p *Deployer) rebaseOpenPRsForRoot(ctx workflow.Context, repo github.Repo) error {
```

Comment:
I don't like how so many things are being shoved in the deployer can you just create a new struct for this and keep all the logic there?